File: Painting_the_Fence.py
class Solution:
    def countWays(self,n,k):
        total = k
        same = 0
        diff = k
        mod = int(1e9) + 7
        for i in range(2, n + 1):
            same = diff
            diff = (total % mod * (k - 1) % mod) % mod
            total = (same + diff) % mod
        return total

File: Painting_the_Fence.java
class Solution{
    long countWays(int n,int k){
        long total = k, same = 0, diff = k, mod = (long)1e9 + 7;
        for(int i = 2; i <= n; i++){
            same = diff;
            diff = (total % mod * (k - 1) % mod) % mod;
            total = (same + diff) % mod;
        }
        return total;
    }
}
